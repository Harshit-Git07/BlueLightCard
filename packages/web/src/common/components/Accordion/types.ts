export interface AccordionProps {
  title: string;
  children: React.ReactNode;
  onClickOpen?: () => void;
}
