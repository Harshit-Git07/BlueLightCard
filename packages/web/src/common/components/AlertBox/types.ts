export interface AlertBoxProps {
  alertType: string;
  title: string;
  description: any;
}

export interface style {
  textColor: string;
  backgroundColor: string;
}

export interface config {
  [key: string]: style;
}
