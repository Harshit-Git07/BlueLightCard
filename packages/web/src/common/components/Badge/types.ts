export type BgColorString = `bg-${string}`;

export type BadgeProps = {
  label: string;
  color: BgColorString;
  size: 'small' | 'large';
};
