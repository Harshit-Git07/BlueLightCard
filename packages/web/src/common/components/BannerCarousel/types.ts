import { CampaignCardProps } from '../CampaignCard/types';

export type BannerCarouselPropers = {
  banners: CampaignCardProps[];
};
