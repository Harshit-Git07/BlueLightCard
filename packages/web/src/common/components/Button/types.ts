import { ThemeVariant } from '@/types/theme';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { MouseEventHandler, PropsWithChildren } from 'react';

export type ButtonProps = PropsWithChildren & {
  type?: 'button' | 'submit' | 'link';
  disabled?: boolean;
  slim?: boolean;
  className?: string;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  onClick?: MouseEventHandler;
};
