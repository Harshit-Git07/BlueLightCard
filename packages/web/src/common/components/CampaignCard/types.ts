export type CampaignCardProps = {
  name: string;
  image: string;
  linkUrl: string;
};
