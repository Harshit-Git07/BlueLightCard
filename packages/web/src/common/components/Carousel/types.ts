export type CarouselProps = {
  children: React.ReactNode[] | React.ReactNode;

  // Display
  hidePillButtons?: boolean;
  loop?: boolean;
  focusCenter?: boolean;

  // Breakpoints
  elementsPerPageDesktop?: number;
  elementsPerPageLaptop?: number;
  elementsPerPageTablet?: number;
  elementsPerPageMobile?: number;

  // Control
  clickToScroll?: boolean;
  showControls?: boolean;
  swipeToScroll?: boolean;

  // Auto play
  autoPlay?: boolean;
  autoPlayIntervalMs?: number;

  // Custom styles
  className?: string;
  focusedClassName?: string;
};
