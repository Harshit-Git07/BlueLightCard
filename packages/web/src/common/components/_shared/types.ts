import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ChangeEventHandler, ForwardedRef, MouseEventHandler, PropsWithChildren } from 'react';

export type InputFieldWrapperProps = PropsWithChildren & {
  icon?: IconDefinition;
  iconRight?: IconDefinition;
  showRightIcon?: boolean;
  showErrorState?: boolean;
  select?: boolean;
  showSuccessState?: boolean;
  onRightIconClick?: MouseEventHandler;
};

export interface InputFieldSharedProps<E, V = string> {
  name?: string;
  error?: boolean;
  success?: boolean;
  value?: V;
  required?: boolean;
  _ref?: ForwardedRef<unknown>;
  onChange?: ChangeEventHandler<E>;
}
