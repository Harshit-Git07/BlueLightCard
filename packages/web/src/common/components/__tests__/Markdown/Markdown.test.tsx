import Markdown from '@/components/Markdown/Markdown';
import { MarkdownProps } from '@/components/Markdown/types';
import renderer from 'react-test-renderer';

describe('Markdown component', () => {
  let props: MarkdownProps;

  it('should render Markdown content', () => {
    const props = {
      content: 'These are the terms and conditions for the above offer.',
    };

    const component = renderer.create(<Markdown {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Markdown with bold text', () => {
    const props = {
      content: 'These are **bold** terms and conditions for the above offer.',
    };

    const component = renderer.create(<Markdown {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Markdown with italic text', () => {
    const props = {
      content: 'These are *italic* terms and conditions for the above offer.',
    };

    const component = renderer.create(<Markdown {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Markdown with strong text', () => {
    const props = {
      content: 'These are __strong__ terms and conditions for the above offer.',
    };

    const component = renderer.create(<Markdown {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Markdown with new lines text', () => {
    const props = {
      content: 'These are \n terms and conditions \n on a new line \n for the above offer.',
    };

    const component = renderer.create(<Markdown {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Markdown with strike through text', () => {
    const props = {
      content: '~~These are terms and conditions on a new line for the above offer.~~',
    };

    const component = renderer.create(<Markdown {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
