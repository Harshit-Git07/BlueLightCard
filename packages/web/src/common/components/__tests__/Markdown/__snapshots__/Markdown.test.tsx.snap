// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown component should render Markdown content 1`] = `
<span>
  These are the terms and conditions for the above offer.
</span>
`;

exports[`Markdown component should render Markdown with bold text 1`] = `
<span>
  These are 
  <strong>
    bold
  </strong>
   terms and conditions for the above offer.
</span>
`;

exports[`Markdown component should render Markdown with italic text 1`] = `
<span>
  These are 
  <em>
    italic
  </em>
   terms and conditions for the above offer.
</span>
`;

exports[`Markdown component should render Markdown with new lines text 1`] = `
<p>
  These are 
 terms and conditions 
 on a new line 
 for the above offer.
</p>
`;

exports[`Markdown component should render Markdown with strike through text 1`] = `
<del>
  These are terms and conditions on a new line for the above offer.
</del>
`;

exports[`Markdown component should render Markdown with strong text 1`] = `
<span>
  These are 
  <strong>
    strong
  </strong>
   terms and conditions for the above offer.
</span>
`;
